pkg/cmd/admin/prune/images.go

```diff
@@ -50,85 +60,138 @@ func NewCmdPruneImages(f *clientcmd.Factory, parentName, name string, out io.Wri
-	cmd.Flags().DurationVar(&cfg.KeepYoungerThan, "keep-younger-than", cfg.KeepYoungerThan, "Specify the minimum age of a build for it to be considered a candidate for pruning.")
-	cmd.Flags().IntVar(&cfg.KeepTagRevisions, "keep-tag-revisions", cfg.KeepTagRevisions, "Specify the number of image revisions for a tag in an image stream that will be preserved.")
-	cmd.Flags().StringVar(&cfg.CABundle, "certificate-authority", cfg.CABundle, "The path to a certificate authority bundle to use when communicating with the managed Docker registries. Defaults to the certificate authority data from the current user's config file.")
-	cmd.Flags().StringVar(&cfg.RegistryUrlOverride, "registry-url", cfg.RegistryUrlOverride, "The address to use when contacting the registry, instead of using the default value. This is useful if you can't resolve or reach the registry (e.g.; the default is a cluster-internal URL) but you do have an alternative route that works.")
+	allBCs, err := osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	err = oserrors.TolerateNotFoundError(err)
+	if err != nil {
+		return err
+	}
 
-	return cmd
+	allBuilds, err := osClient.Builds(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	err = oserrors.TolerateNotFoundError(err)
+	if err != nil {
+		return err
+	}
+
+	allDCs, err := osClient.DeploymentConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return err
+	}
+
+	options := prune.ImageRegistryPrunerOptions{
+		KeepYoungerThan:  o.KeepYoungerThan,
+		KeepTagRevisions: o.KeepTagRevisions,
+		Images:           allImages,
+		Streams:          allStreams,
+		Pods:             allPods,
+		RCs:              allRCs,
+		BCs:              allBCs,
+		Builds:           allBuilds,
+		DCs:              allDCs,
+		DryRun:           o.Confirm == false,
+		RegistryClient:   registryClient,
+		RegistryURL:      o.RegistryUrlOverride,
+	}
+
+	o.Pruner = prune.NewImageRegistryPruner(options)
+
+	return nil
+}
+
+// Validate the options for prune images
+func (o *PruneImagesOptions) Validate() error {
+	if o.Pruner == nil && o.Confirm {
+		return errors.New("an image pruner needs to be specified")
+	}
+	if o.Client == nil {
+		return errors.New("a client needs to be specified")
+	}
+	if o.Out == nil {
+		return errors.New("a writer needs to be specified")
+	}
+	return nil
+}
+
+// RunPruneImages runs the prune images cli command
+func (o *PruneImagesOptions) RunPruneImages() error {
+	// this tabwriter is used by the describing*Pruners below for their output
+	w := tabwriter.NewWriter(o.Out, 10, 4, 3, ' ', 0)
+	defer w.Flush()
+
+	imagePruner := &describingImagePruner{w: w}
+	imageStreamPruner := &describingImageStreamPruner{w: w}
+	layerPruner := &describingLayerPruner{w: w}
+	blobPruner := &describingBlobPruner{w: w}
+	manifestPruner := &describingManifestPruner{w: w}
+
+	switch o.Confirm {
```

Comment:
Actually this is very "nity" since I will still have to use an else which I hate. Leaving this as is:)